k2/python/csrc/torch/arc.cu

```diff
@@ -0,0 +1,46 @@
+/**
+ * @brief python wrappers for Arc.
+ *
+ * @copyright
+ * Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include <string>
+
+#include "k2/csrc/fsa.h"
+#include "k2/python/csrc/torch/arc.h"
+
+namespace k2 {
+
+static void PybindArcImpl(py::module &m) {
+  using PyClass = Arc;
+  py::class_<PyClass> pyclass(m, "Arc");
+  pyclass.def(py::init<>());
+  pyclass.def(py::init<int32_t, int32_t, int32_t, float>(),
+              py::arg("src_state"), py::arg("dest_state"), py::arg("symbol"),
+              py::arg("score"));
+
+  pyclass.def_readwrite("src_state", &PyClass::src_state)
+      .def_readwrite("dest_state", &PyClass::dest_state)
+      .def_readwrite("symbol", &PyClass::symbol)
+      .def_readwrite("score", &PyClass::score);
+
+  pyclass.def("__str__", [](const PyClass &self) -> std::string {
+    std::ostringstream os;
+    os << self;
+    return os.str();
+  });
+
+  m.def("float_as_int",
```

Comment:
I think it would be great if we can define such function (i.e. reinterpret int as float or vice versa ) for a tensor (e.g. for arcs' weight in an Fsa which is a n*1 tensor, or even for an Fsa), as we usually would access a tensor instead of a particular value in the tensor.

Reply:
Done.

Reply:
With these kinds of things I think the documentation might be a problem.  When a Python programmer sees this, who doesn't understand C++, what will they make of it?  It's even a little confusing to me, e.g. how would this be invoked?  As _k2.float_as_int(1.0) ->  [some integer] ?

Reply:
Also, I was thinking of separating the FSA-related things into a different namespace k2fsa, instead of k2, at the python level.  I don't know if you guys think that is a bad idea though.
Of course we may still use the same internal namespace, _k2, for all the C++-exported things.

Reply:
They are invoked like this:

https://github.com/k2-fsa/k2/blob/a6b25e8daf33d7ab04999109da6772d2b17c62c1/k2/python/tests/fsa_test.py#L58-L66

An arc contains 3 integers and 1 float but it is saved as a torch::Tensor with dtype==torch.int32. We want to reinterpret the last integer as a float.

When `int_as_float` is applied to a tensor, it is invoked like this:

https://github.com/k2-fsa/k2/blob/a6b25e8daf33d7ab04999109da6772d2b17c62c1/k2/python/tests/fsa_test.py#L68-L76

Reply:
I was thinking those APIs would be just for internal usage, we can expose functions later such as `Fsa.get_weights`, 'Fsa.create_from_arc_and_weights' which may be called directly by the users, based on below reinterpretation version for tensor.

Reply:
There is a `view` method in `numpy`: https://numpy.org/doc/stable/reference/generated/numpy.ndarray.view.html

We should probably rename them to `view`.

PyTorch users have this kind of need, see https://discuss.pytorch.org/t/reinterpret-pytorch-array-as-a-different-dtype/24256

Unfortunately, PyTorch does support this kind of operation.

Reply:
> When a Python programmer sees this, who doesn't understand C++, what will they make of it?

If we rename them to `view`, I think a Python programmer who is familiar with `numpy` knows the meaning
of the method once he/she sees it.

Reply:
> Also, I was thinking of separating the FSA-related things into a different namespace k2fsa, instead of k2, at the python level. 

this is fine with me.

Reply:
> Also, I was thinking of separating the FSA-related things into a different namespace k2fsa, instead of k2, at the python level. I don't know if you guys think that is a bad idea though.
 
Did you mean we suppose user just call Fsa-related code at python level? no Ragged and Array? If user still need to use Ragged and Array, I don't see the necessity to just put fsa code into a new namespace (as user will see two global namespace? i.e. k2 and k2fsa?)

Reply:
`torch.Tensor` has a `view` method: https://pytorch.org/docs/stable/tensors.html#torch.Tensor.view
However, it only supports reshaping and does not allow to change the underlying data type.

Reply:
Now, these methods are used only **internally**.